src/scene/particle-container/shared/Particle.ts

```diff
@@ -0,0 +1,144 @@
+import { Color } from '../../../color/Color';
+import { Texture } from '../../../rendering/renderers/shared/texture/Texture';
+
+import type { ColorSource } from '../../../color/Color';
+
+/**
+ * Represents a particle with properties for position, scale, rotation, color, and texture.
+ * @property {number} x - The x-coordinate of the particle.
+ * @property {number} y - The y-coordinate of the particle.
+ * @property {number} scaleX - The scale factor in the x-axis.
+ * @property {number} scaleY - The scale factor in the y-axis.
+ * @property {number} anchorX - The x-coordinate of the anchor point.
+ * @property {number} anchorY - The y-coordinate of the anchor point.
+ * @property {number} rotation - The rotation of the particle in radians.
+ * @property {number} color - The color of the particle as a hexadecimal number.
+ * @property {Texture} texture - The texture of the particle.
+ */
+export interface IParticle
+{
+    x: number;
+    y: number;
+    scaleX: number;
+    scaleY: number;
+    anchorX: number;
+    anchorY: number;
+    rotation: number;
+    color: number;
+    texture: Texture;
+}
+
+export type ParticleOptions = Omit<Partial<IParticle>, 'color'> & {
+    texture: Texture
+    tint?: number;
```

Comment:
This could be more flexible:

```suggestion
    tint?: ColorSource;
```